### DISCUSSION TITLE
How do we use a constant's value as a lingui id?

### DISCUSSION THREAD

**Discussion by santiagopuentep** [lingui/js-lingui#1112] [Q&A] [answered]
I have constant like so: `const SOME_ERROR = "some-error";` and I want `some-error` to be the actual lingui id, so that we can add there the error message and have it say something different in every language.

So, the result we want is: `msgid "some-error"` in the .po file.

Is there a way to do it?

Thanks!

**Comment by manuganji** [comment]
So far, I am using it like so.

```
import { t  } from "@lingui/macro";


export const PAYMENT_TYPES = {
  CASH: {
    code: "cash",
    label: t`Cash`,
    },
}
 ```
 
Later, I can use it like so.

```
import {i18n} from "@lingui/core";


{i18n._(PAYMENT_TYPES.CASH.label)}
```

So far, it's only working in `DEV`  for me.

**Reply by semoal** [reply]
For constant usages, it's recommended to use `defineMessage` instead of `t`.

**Comment by santiagopuentep** [comment, marked as answer]
We ended up using it like this:

```
t`my-text-constant`
const MI_TEXT_CONSTANT = "my-text-constant";
```

The id generation next to the actual constant, with comments making sure that devs always keep the macro and the actual string the same.

This way Lingui generates the id, and the constant is set with the literal text and not with the translation.

**Reply by timofei-iatsenko** [reply]
That also supported by extractor
```ts
const MI_TEXT_CONSTANT =  /*i18n*/ "my-text-constant";
```

And next: 

```js
import {i18n} from "@lingui/core"

// extractor should not complain about variable here
i18n._(MI_TEXT_CONSTANT)
```

**Reply by santiagopuentep** [reply]
@thekip awesome. Thanks!
